webapp/channels/src/actions/websocket_actions.jsx

```diff
@@ -1747,6 +1751,22 @@ function handleUpsertDraftEvent(msg) {
     };
 }
 
+function handleUpsertScheduledPostEvent(msg) {
+    return async (doDispatch) => {
+        const scheduledPost = JSON.parse(msg.data.scheduledPost);
+        const state = getState();
+        const teamId = getTeamIdFromPost(state, scheduledPost);
```

Comment:
So `getTeamIdFromPost` expects a `Post` type as second param, while this works fine because both Post and ScheduledPost have a `channel_id` field, technically this is wrong specially if we convert this file to support TypeScript.

I see that the function `getTeamIdFromPost` is being used only in one more place, so maybe you can update that function to receive just `channelId: string` param instead of `post: Post` param, that way its agnostic of post, scheduled post or even draft.